tests/model_fields/tests.py

```diff
@@ -661,6 +689,15 @@ def test_backend_range_validation(self):
             }
             with self.assertRaisesMessage(ValidationError, expected_message):
                 instance.full_clean()
+            try:
+                with transaction.atomic():
+                    instance.save()
+            except (DataError, IntegrityError):
+                pass  # backends such as Postgres don't fail silently
+            else:
+                qs = self.model.objects.filter(value__lte=min_value)
```

Comment:
It will be more friendly to third-party database backends if we add a feature flag for the behavior (django/db/backends/base/features.py). Maybe it's enough to include the `test_backend_range` test for this pull request (to simplify the backport) and save the other additional asserts and feature flags for a separate commit?